apollo-compiler/src/test/graphql/com/example/arguments_complex/TestQuery.kt

```diff
@@ -72,6 +75,32 @@ data class TestQuery(
   @Throws(IOException::class)
   override fun parse(source: BufferedSource): Response<Data> = parse(source, DEFAULT)
 
+  override fun composeRequestBody(scalarTypeAdapters: ScalarTypeAdapters): ByteString =
+      OperationRequestBodyComposer.compose(
+    operation = this,
+    autoPersistQueries = false,
+    withQueryDocument = true,
+    scalarTypeAdapters = scalarTypeAdapters
+  )
+
+  override fun composeRequestBody(): ByteString = OperationRequestBodyComposer.compose(
+    operation = this,
+    autoPersistQueries = false,
+    withQueryDocument = true,
+    scalarTypeAdapters = DEFAULT
+  )
+
+  override fun composeRequestBody(
+    autoPersistQueries: Boolean,
+    withQueryDocument: Boolean,
+    scalarTypeAdapters: ScalarTypeAdapters
+  ): ByteString = OperationRequestBodyComposer.compose(
+    operation = this,
+    autoPersistQueries = autoPersistQueries,
+    withQueryDocument = withQueryDocument,
+    scalarTypeAdapters = scalarTypeAdapters
+  )
```

Comment:
Does this need to be in the generated models ?  What about just having that as an extension function in kotlin ? That's less codegen code to maintain and it'll make it easier to navigate the generated models ?

Reply:
I see you concern but I think it should be part of generated models for 2 reasons:

- as it's a symmetric to the generated method implementation of `Operation.parse` 
- and it's more nicer API from the iOS user perspective

Reply:
> as it's a symmetric to the generated method implementation of Operation.parse

Could `Operation.parse` be made an extension method ? 

> and it's more nicer API from the iOS user perspective

Indeed. Although I guess a lot of users might still consume this API from Kotlin but for those consuming it in Swift, that'll be way better 👍

Reply:
Yeah we could make `parse` as extension function but again for Swift consumers it won't be very nice experience.